cpp/include/cudf/column/column_device_view.cuh

```diff
@@ -268,11 +268,11 @@ class alignas(16) column_device_view_base {
   size_type _offset{};               ///< Index position of the first element.
                                      ///< Enables zero-copy slicing
 
-  column_device_view_base(data_type type,
-                          size_type size,
-                          void const* data,
-                          bitmask_type const* null_mask,
-                          size_type offset)
+  CUDF_HOST_DEVICE column_device_view_base(data_type type,
```

Comment:
strictly speaking, this macro is really only necessary in a `.hpp` file that is expected to work in both host and device code TUs.

Reply:
> strictly speaking, this macro is really only necessary in a .hpp file that is expected to work in both host and device code TUs.

I agree that it's not necessary in `cuh`. I made this suggestion because not everyone will think about that and people will invariably copy-paste, so uniformly using `CUDF_HOST_DEVICE` helps reduce the chance of future errors and leaves one less thing for developers to think about. It's a minor suggestion though, if you prefer to use it more surgically that's OK with me.